catalog/redis-bucket/Kptfile

```diff
@@ -11,6 +11,16 @@ info:
 
     This blueprint demonstrates multiple unrelated resources sharing a package.
 pipeline:
+  # File gcloud-config.yaml is autogenerated by `kpt pkg get` locally.
   mutators:
-    - image: gcr.io/kpt-fn/apply-setters:v0.1
-      configPath: setters.yaml
+    - image: gcr.io/kpt-fn-demo/set-default-name:v0.2-yuwenma
```

Comment:
What's the idea behind `set-default-name` vs prefix?

Reply:
"prefix function" --> Do you mean the [`ensure-name-substring`](https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/master/functions/go/ensure-name-substring)? 

The `ensure-name-substring` add prefix or suffix to a meta name but cannot replace the whole name. Also, it does not support nameReference the way this blueprint needs. For example, this blueprint expects the GVK `RedisInstance` has `spec.displayName` matching its meta name. `ensure-name-string` won't make the matches, it checks if the `spec.displayName's meta data matches the RedisInstance's meta name (which is checking if it matches itself), and adds prefix/suffix to it.  See example at the bottom.

When using the  `ensure-name-substring`, users have to provide a new resource `ConfigMap` or `EnsureNameSubstring` that following the schema, and it does not fit into our gcloudconfig.yaml for all functionConfig case.

`redis.yaml` **before** running the function
```yaml
kind:  RedisInstance
metadata:
   name: sample-instance
spec:
   displayName: mis-spell-instance
```
Assume somehow users can successfully configure the  `ensure-name-substring` functionConfig in their local. The best similar config will be: 
```yaml
apiVersion: v1
kind: EnsureNameSubstring # cannot use ConfigMap format because the "additionalNameField
metadata:
  name: my-fn-config
substring: dev-
editMode: prepend
additionalNameFields:
  - path: spec/displayName
    kind: EnsureNameSubstring
```
`redis.yaml` **after** running the function
```yaml
kind:  RedisInstance
metadata:
   name: dev-sample-instance # add prefix
spec:
   displayName: dev-mis-spell-instance # add prefix. but "mis-spell-instance" are not corrected.
```

Reply:
ICYMI, go/cad-blueprints has a section talking about the problems of existing `ensure-name-substring`